bin/fba_cmx

```diff
@@ -0,0 +1,817 @@
+if (args.flavor in ['dithprec','dithlost']) & (tile_in_desi==0):
+    mydirs['targ']= os.path.join(path_to_targets, 'gaiadr2', args.dtver, 'targets/cmx/resolve/supp')
+else:
+    mydirs['targ']= os.path.join(path_to_targets, args.dr,args.dtver,'targets/cmx/resolve/no-obscon')
+mydirs['sky']     = os.path.join(path_to_targets, args.dr, args.dtver, 'skies')
+mydirs['skysupp'] = os.path.join(path_to_targets, 'gaiadr2', args.dtver, 'skies-supp')
+mydirs['gfa']     = os.path.join(path_to_targets, args.dr, args.dtver, 'gfas')
+for key in mydirs.keys():
+    log.info('{:.1f}s\tdirectory for {}: {}'.format(time()-start,key,mydirs[key]))
+log.info('{:.1f}s\tdirectory for svn tiles: {}'.format(time()-start, path_to_svn_tiles))
+
+
+
+# AR extra tileid if ndither>0; switching to np.array() in all cases
+tileids = np.array([args.tileid+i for i in range(1+fdict['ndither'])])
+log.info('{:.1f}s\twill process {} tiles with tileid={}'.format(time()-start,1+fdict['ndither'],','.join([str(tileid) for tileid in tileids])))
+# AR safe tileids
+prev_fns   = [fn.split('/')[-1] for fn in glob(os.path.join(path_to_svn_tiles, '???/fiberassign-??????.fits'))]
+
+
+new_fns    = ['fiberassign-{:06d}.fits'.format(tid) for tid in tileids]
+if (np.in1d(new_fns,prev_fns).sum()>0):
+    log.error('{:.1f}s\tsome of {} files already exist; exiting'.format(time()-start,','.join(new_fns)))
+    sys.exit()
+
+
+
+# AR printing settings
+tmpstr  = ' , '.join([kwargs[0]+'='+str(kwargs[1]) for kwargs in args._get_kwargs()])
+log.info('{:.1f}s\targs: {}'.format(time()-start,tmpstr))
+tmpstr  = ' , '.join([key+'='+str(fdict[key]) for key in fdict.keys()])
+log.info('{:.1f}s\tfdict: {}'.format(time()-start,tmpstr))
+
+
+
+# AR copied from make_mtl()
+mtldatamodel = np.array([], dtype=[
+    ('RA', '>f8'), ('DEC', '>f8'), ('PARALLAX', '>f4'),
+    ('PMRA', '>f4'), ('PMDEC', '>f4'), ('REF_EPOCH', '>f4'),
+    ('DESI_TARGET', '>i8'), ('BGS_TARGET', '>i8'), ('MWS_TARGET', '>i8'),
+    ('SCND_TARGET', '>i8'), ('TARGETID', '>i8'),
+    ('SUBPRIORITY', '>f8'), ('OBSCONDITIONS', 'i4'),
+    ('PRIORITY_INIT', '>i8'), ('NUMOBS_INIT', '>i8'), ('PRIORITY', '>i8'),
+    ('NUMOBS', '>i8'), ('NUMOBS_MORE', '>i8'), ('Z', '>f8'), ('ZWARN', '>i8'),
+    ('TIMESTAMP', 'S19'), ('VERSION', 'S14'), ('TARGET_STATE', 'S15')
+    ])
+
+
+
+# AR extra-hdu for dithering
+extradatamodel = np.array([], dtype=[
+    ('UNDITHER_RA','>f8'),('UNDITHER_DEC','>f8'),('TARGETID', '>i8')
+    ])
+
+
+
+# AR obscon
+if   (args.obscon=='dark,bright'): obscon = 'DARK|GRAY|BRIGHT'
+elif (args.obscon=='dark'):        obscon = 'DARK|GRAY'
+else:                              obscon = 'BRIGHT' # AR bright
+log.info('{:.1f}s\tsetting obscon = {}'.format(time()-start,obscon))
+
+
+
+# AR get matching index for two np arrays, those should be arrays with unique values, like id
+# AR https://stackoverflow.com/questions/32653441/find-indices-of-common-values-in-two-arrays
+# AR we get: A[maskA] = B[maskB]
+def unq_searchsorted(A,B):
```

Comment:
If we had a `main()`, this would go outside of it.  Or if it is generally useful we could move it to the utils package or into `desiutils`.